core/include/detray/propagator/rk_stepper.hpp

```diff
@@ -71,9 +71,14 @@ class rk_stepper final
 
         /// stepping data required for RKN4
         struct {
-            vector3 b_first, b_middle, b_last;
-            vector3 k1, k2, k3, k4;
-            array_t<scalar, 4> k_qop;
+            vector3 b_first{0.f, 0.f, 0.f};
```

Comment:
I don't think, you have to put all of the ```0``` in here, empty braces *should* do zero-initialization, especially on the array:
```array_t<scalar, 4> k_qop{};```